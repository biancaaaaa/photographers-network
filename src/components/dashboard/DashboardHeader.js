import React from "react";
import { Link } from "react-router-dom";
import { Button } from "../Button";
import PropTypes from "prop-types";
import AvatarInput from "../../contents/AvatarInput";

export const DashboardHeader = ({
  children,
  links,
  user,
  linkHandler,
  type,
  updateUserInfo
}) => {
  let profilePath = `/profile/${user.uid}`;
  return (
    <div>
      <div className="gb-card-7-wrapper">
        <div
          style={{ flexDirection: "column" }}
          className="gb-card-7-height gb-background-primary"
        >
          <AvatarInput
            uid={user.uid}
            userAvatar={user.photoURL}
            updateUserInfo={updateUserInfo}
          />
          <div className="card-7-shadow-overlay" />
          <div className="card-7-content">
            <h1 className="gb-title-xx-large gb-text-white gb-margin-bottom-40 gb-text-align-center">
              {children}
            </h1>
            <div className="gb-display-flex">
              {links.map((link, key) => {
                return (
                  <Button
                    key={key}
                    clickHandler={() => linkHandler(link.name, type)}
                    classes="gb-btn gb-btn-small gb-btn-outlined gb-margin-right-16"
                  >
                    {link.name}
                  </Button>
                );
              })}
              <Link
                to={profilePath}
                className="gb-btn gb-btn-small gb-btn-outlined gb-margin-right-16"
              >
                Profile
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

DashboardHeader.propTypes = {
  children: PropTypes.array.isRequired,
  links: PropTypes.array,
  uid: PropTypes.string,
  linkHandler: PropTypes.func,
  type: PropTypes.string
};
