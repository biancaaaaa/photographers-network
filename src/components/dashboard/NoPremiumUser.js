import React from "react";
import { PropTypes } from "prop-types";
import WithModal from "../../RenderProp/WithModal";
import fire from "../../config/Fire";

export default class NoPremiumUser extends React.Component {
  state = {
    buttonStatus: "Become Premium"
  };
  static propTypes = {
    user: PropTypes.object.isRequired,
    updateUserInfo: PropTypes.func.isRequired
  };

  makeUserPremium = () => {
    if (this.state.buttonStatus !== "Become Premium") {
      return;
    }
    this.setState(() => ({
      buttonStatus: "Loading..."
    }));
    fire
      .database()
      .ref("users")
      .child(this.props.user.uid)
      .update(
        {
          premium: true
        },
        err => {
          if (err) {
            this.setState(() => ({
              buttonStatus: "Error"
            }));
          } else {
            this.setState(() => ({
              buttonStatus: "Done!"
            }));
            this.props.updateUserInfo({ premium: true });
          }
        }
      );
  };

  render() {
    const { buttonStatus } = this.state;
    return (
      <React.Fragment>
        <p>For a portofolio you have to be premium...</p>
        <WithModal closeItemClass="close-modal">
          {({ showModal }) => (
            <React.Fragment>
              <h2>Become premium</h2>
              <div
                style={{ display: showModal ? "flex" : "none" }}
                className="modal-become-premium close-modal"
              >
                <div className="modal-inner-box">
                  <h2>
                    If you become premium it will be awsome trust me, you will
                    not regret it!..
                  </h2>
                  <div
                    onClick={this.makeUserPremium}
                    className={`become-premium-button ${
                      buttonStatus == "Done!" ? "close-modal" : ""
                    }`}
                  >
                    {buttonStatus}
                  </div>
                </div>
              </div>
            </React.Fragment>
          )}
        </WithModal>
      </React.Fragment>
    );
  }
}
