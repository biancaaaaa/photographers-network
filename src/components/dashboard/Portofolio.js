import React from "react";
import { PropTypes } from "prop-types";
import { PortofolioGallery } from "../PortofolioGallery";
import WithModal from "../../RenderProp/WithModal";
import UploadPhotoToPortofolio from "../../contents/UploadPhotoToPortofolio";

export default class Portofolio extends React.Component {
  static propTypes = {
    user: PropTypes.object,
    updateUserInfo: PropTypes.func.isRequired
  };
  render() {
    const { user, updateUserInfo } = this.props;
    return (
      <React.Fragment>
        {user.portofolio.length > 0 ? (
          <h2>Portofolio.</h2>
        ) : (
          <h2>You have no photos, add your first one in here.</h2>
        )}

        <WithModal className="portofolio-add">
          {({ showModal, closeModalListener }) => (
            <UploadPhotoToPortofolio
              closeModalListener={closeModalListener}
              ref={element => (this.uploadPhotoRef = element)}
              inputChangeHandler={this.inputChangeHandler}
              user={this.props.user}
              updateUserInfo={updateUserInfo}
              showModal={showModal}
            />
          )}
        </WithModal>
        {user.portofolio.length > 0 && (
          <PortofolioGallery photosList={user.portofolio} />
        )}
      </React.Fragment>
    );
  }
}
