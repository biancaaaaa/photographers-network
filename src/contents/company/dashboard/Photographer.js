import React from "react";
import {Link} from "react-router-dom";
import PropTypes from 'prop-types';
import {Avatar} from "../../../components/label/Avatar";

export const Photographer = ({profileImageUrl, firstName, lastName, locations, description, uid}) => (
  <li className="user-li-card">
    <div className="gb-display-flex">
      <Avatar userImageUrl={profileImageUrl}/>
      <div>
        <h1 className="uppercase">{firstName} {lastName}</h1>
        <h4 className="small-heading">{Object.values(locations)[0].city}</h4>
      </div>
    </div>
    <p className="description">
      {description}
    </p>
    <Link className="gb-btn gb-btn-small gb-btn-primary gb-btn-white pink-border"
          to={`/profile/${uid}`}>
      Profile
    </Link>
  </li>
);

Photographer.propTypes = {
  profileImageUrl: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  description: PropTypes.string,
  locations: PropTypes.object.isRequired,
  uid: PropTypes.string.isRequired
};