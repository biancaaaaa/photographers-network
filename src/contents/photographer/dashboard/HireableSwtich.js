import React from "react";
import { switchHireable } from "../../../redux/actions/photographer-actions";
import { connect } from "react-redux";

class HireableSwitch extends React.Component {
  state = {
    loading: false,
    error: false
  };

  switchHireableStatus = () => {
    const { hireable, uid, switchHireable } = this.props;
    this.setState({ loading: true });

    //Call the action that lives in user-actions, wait for the promise
    switchHireable(!hireable, uid)
      .then(() => {
        this.setState({
          loading: false,
          error: false
        });
      })
      .catch(() =>
        this.setState({
          loading: false,
          error: true
        })
      );
  };

  render() {
    const { hireable } = this.props;
    const { loading, error } = this.state;

    return (
      <div className="hireable-option-changer">
        <label>
          You are {!hireable && "not"} hireable!
          <input
            onChange={this.switchHireableStatus}
            type="checkbox"
            checked={hireable}
            disabled={loading}
          />
          <Slider on={hireable} waiting={loading} />
        </label>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const firebaseData = state.firebase;
  return {
    hireable: firebaseData.profile.hireable,
    uid: firebaseData.auth.uid
  };
};

const mapDispatchToProps = dispatch => ({
  switchHireable: (to, uid) => dispatch(switchHireable(to, uid))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HireableSwitch);

const Slider = ({ on, waiting }) => (
  <div style={{ opacity: waiting ? 0.5 : 1 }} className="mySlider-track">
    <div className={`mySlider-tumb ${on ? "on" : ""}`} />
  </div>
);
