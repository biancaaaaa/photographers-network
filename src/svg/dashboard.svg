<svg width="43px" height="37px" viewBox="0 0 43 37" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <g id="Mobile" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="16-Navigation" transform="translate(-74.000000, -87.000000)">
            <g id="Icon" transform="translate(70.000000, 80.000000)">
                <g id="Group" transform="translate(4.000000, 7.000000)" stroke="rgba(255, 255, 255, 0.5)" stroke-width="2">
                    <rect id="Oval" x="1" y="1" width="6.5106383" height="6.83888889"></rect>
                    <rect id="Oval" x="1" y="15.06" width="6.5106383" height="6.83888889"></rect>
                    <rect id="Oval" x="1" y="29.1611111" width="6.5106383" height="6.83888889"></rect>
                    <path d="M15.3191489,5.30333333 L41.3191489,5.30333333" id="Line" stroke-linecap="round"></path>
                    <path d="M15.3191489,19.3633333 L41.3191489,19.3633333" id="Line" stroke-linecap="round"></path>
                    <path d="M15.3191489,32.5805556 L41.3191489,32.5805556" id="Line" stroke-linecap="round"></path>
                </g>
                <rect id="bounds" x="0" y="0" width="50" height="50"></rect>
            </g>
        </g>
    </g>
</svg>