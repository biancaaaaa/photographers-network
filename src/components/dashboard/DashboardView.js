// dependencies
import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

// components
import { DashboardHeader } from "./DashboardHeader";

// contents
import MyJobOffers from "../../contents/company/my-jobs/MyJobOffers";
import AppliedJobs from "../../contents/photographer/applied-jobs/AppliedJobs";
import NoPremiumUser from "./NoPremiumUser";
import Portofolio from "./Portofolio";
import HireableSwitch from "../../contents/photographer/dashboard/HireableSwtich";

export const DashboardView = ({
  profile,
  auth,
  type,
  linkHandler,
  activeComponent,
  headerLinks
}) => {
  let currentComponent = "";

  //TODO: maybe find a better solultion than the switch?
  switch (activeComponent) {
    case "Home":
      currentComponent =
        type === "photographer" ? (
          // user is photographer
          <React.Fragment>
            {profile.premium ? (
              // user is premium
              <Portofolio user={profile} />
            ) : (
              // user is not premium
              <NoPremiumUser user={profile} />
            )}
            <Link to="/jobs" className="gb-btn gb-btn-medium gb-btn-primary">
              Search for jobs
            </Link>
            <HireableSwitch user={profile} />
          </React.Fragment>
        ) : (
          // user is company
          <React.Fragment>
            <Link
              to="../createJob"
              className="gb-btn gb-btn-medium gb-btn-primary"
            >
              Create job offer
            </Link>
            <Link
              to="/search-photographers"
              className="gb-btn gb-btn-medium gb-btn-primary"
            >
              Search for photographers
            </Link>
          </React.Fragment>
        );
      break;
    case "Applied Jobs":
      currentComponent = <AppliedJobs user={profile} auth={auth} />;
      break;
    case "My Jobs":
      currentComponent = <MyJobOffers user={profile} auth={auth} />;
      break;
    default:
      currentComponent = <div>No fitting component!</div>;
      break;
  }

  return (
    <div>
      <DashboardHeader
        type={type}
        links={headerLinks}
        profile={profile}
        auth={auth}
        linkHandler={linkHandler}
      >
        {profile.type === "photographer" ? (
          <span>Welcome {`${profile.firstName} ${profile.lastName}`}!</span>
        ) : (
          <span>Welcome {`${profile.companyName}`}!</span>
        )}
      </DashboardHeader>
      {currentComponent}
    </div>
  );
};

DashboardView.propTypes = {
  type: PropTypes.string.isRequired,
  linkHandler: PropTypes.func,
  activeComponent: PropTypes.string,
  headerLinks: PropTypes.array
};
