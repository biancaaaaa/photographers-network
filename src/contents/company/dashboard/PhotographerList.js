import React from "react";
import {connect} from "react-redux";
import {compose} from "redux";
import {isLoaded, isEmpty, firestoreConnect} from "react-redux-firebase";
import {Photographer} from "./Photographer";

const PhotographerList = ({photographers}) => {
  if (!isLoaded(photographers))
    return <p className="dashboard-container loading">Loading...</p>;
  if (isEmpty(photographers))
    return <p className="dashboard-container">No photographers</p>;
  return (
    <ul>
      {
        photographers.map(photographer =>
          <Photographer key={photographer.id} {...photographer}/>
        )
      }
    </ul>
  );
};

const mapStateToProps = state => ({
  photographers: state.firestore.ordered.photographers
});

export default compose(
  firestoreConnect([
    {
      collection: "users",
      where: [["type", "==", "photographer"]],
      storeAs: "photographers"
    }
  ]),
  connect(mapStateToProps)
)(PhotographerList);