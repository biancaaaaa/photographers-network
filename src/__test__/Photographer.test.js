import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import {BrowserRouter as Router} from "react-router-dom";
import {PhotographerList} from '../contents/company/dashboard/PhotographerList';
import {Photographer} from "../contents/company/dashboard/Photographer";

import makeUser from "./mockUser";

describe("Photographer tests", () => {
  let photographer = makeUser({type: "photographer"});
  const shallowRender = () => shallow(<PhotographerList photographers={[photographer]}/>);

  describe("renders correctly with one photographer passed", () => {
    const shallowPhotographerList = shallowRender();
    it("renders one photographer", () => {
      expect(shallowPhotographerList.find(Photographer).length).toBe(1);
    });

    it("renders right content of photographer", () => {
      const tree = renderer
        .create(
          <Router>
            <Photographer
              {...photographer}
            />
          </Router>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("renders loading correctly", () => {
    const shallowPhotographerList = shallowRender();
    shallowPhotographerList.setProps({photographers: undefined});
    it("shows loading, when data is undefined", () => {
      expect(shallowPhotographerList.find(".loading").length).toBe(1);
    });
  });

  describe("renders empty list correctly", () => {
    const shallowPhotographerList = shallowRender();
    shallowPhotographerList.setProps({photographers: null});
    it("shows element for no photographers", () => {
      expect(shallowPhotographerList.find(".no-data").length).toBe(1);
    });
  });
});
