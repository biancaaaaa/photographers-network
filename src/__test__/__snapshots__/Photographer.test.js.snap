// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photographer tests renders correctly with one photographer passed renders right content of photographer 1`] = `
<li
  className="user-li-card"
>
  <div
    className="gb-display-flex"
  >
    <a
      className="profile-link"
      href="#"
    >
      <img
        alt="avatar"
        className="gb-avatar gb-avatar-medium"
        src="e.jpg"
      />
    </a>
    <div>
      <h1
        className="uppercase"
      >
        S
         
        S
      </h1>
      <h4
        className="small-heading"
      >
        S
      </h4>
    </div>
  </div>
  <p
    className="description"
  />
  <a
    className="gb-btn gb-btn-small gb-btn-primary gb-btn-white pink-border"
    href="/profile/zzz"
    onClick={[Function]}
  >
    Profile
  </a>
</li>
`;
